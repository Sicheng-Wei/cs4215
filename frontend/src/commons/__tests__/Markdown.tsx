import { mount } from 'enzyme';
import { Chapter, Variant } from 'js-slang/dist/types';

import Markdown from '../Markdown';
import { generateSourceIntroduction } from '../utils/IntroductionHelper';

const mockProps = (sourceChapter: Chapter, sourceVariant: Variant) => {
  return {
    content: generateSourceIntroduction(sourceChapter, sourceVariant),
    openLinksInNewWindow: true
  };
};

test('Markdown page renders correctly', () => {
  const app = <Markdown {...mockProps(Chapter.SOURCE_1, Variant.DEFAULT)} />;
  const tree = mount(app);
  expect(tree.debug()).toMatchSnapshot();
});
