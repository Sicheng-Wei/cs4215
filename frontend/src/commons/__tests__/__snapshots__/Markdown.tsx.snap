// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown page renders correctly 1`] = `
"<Memo(Markdown) content=\\"\\\\nWelcome to the Source Academy playground!\\\\n\\\\nThis is a stripped down version of the Source Academy for the purpose of demonstrating an implementation\\\\nof a subset of the _C_ programming language. Made by [Geng Ning](https://github.com/zognin) &\\\\n[Ian](https://github.com/ianyong) for CS4215 Programming Language Implementation in AY2022/23 Semester 2.\\\\n\\\\nYou have chosen the sublanguage _C_ (although you never really had a choice).\\\\n\\\\nIn the editor on the left, you can use the [_Ace keyboard shortcuts_](https://github.com/ajaxorg/ace/wiki/Default-Keyboard-Shortcuts) \\\\nand also the [_Source Academy keyboard shortcuts_](https://github.com/source-academy/frontend/wiki/Source-Academy-Keyboard-Shortcuts).\\\\n\\\\n\\" openLinksInNewWindow={true}>
  <div className=\\"md bp4-running-text\\" dangerouslySetInnerHTML={{...}} />
</Memo(Markdown)>"
`;
